### PR TITLE
Register the necessary variables for micro-climate analysis in geotechnics

### LINKED ISSUE #11849: [GeoMechanicsApplication] Add variables for micro-climate
### Description

**This feature is fully implemented.**

Branch: https://github.com/KratosMultiphysics/Kratos/tree/geo/10372-HeatTransfer-temp-branch

List of variables necessary for micro-climate boundary is registerd in the GeoMechanicsApplication.

- applications/GeoMechanicsApplication\geo_mechanics_application.cpp -> Lines 356 to 367

- applications\GeoMechanicsApplication\geo_mechanics_application_variables.cpp -> lines 46 to 57

-------------------------------------------------------------------------------------------------------------------------

The variables registered in the lines below **are not included in this PR**. The intention is to add them later:

- applications\GeoMechanicsApplication\custom_python\geo_mechanics_python_application.cpp -> lines 61 to 65

- applications\GeoMechanicsApplication\python_scripts\geomechanics_solver.py -> lines 358 to 362

### Acceptance Criteria

**Given** the geomechanics application
**When** the application checks the availability of the variables,
**Then** there is no exception

### PR DESCRIPTION
**Pull request by indigocoral**
**📝 Description**
Register the necessary variables for micro-climate analysis in geotechnics

**🆕 Changelog**
The variables registered are:
- "AIR_TEMPERATURE"
- "SOLAR_RADIATION" 
- "AIR_HUMIDITY"
- "PRECIPITATION"
- "WIND_SPEED"
- "A1_COEFFICIENT"
- "A2_COEFFICIENT"
- "A3_COEFFICIENT"
- "ALPHA_COEFFICIENT"
- "QF_COEFFICIENT"
- "SMIN_COEFFICIENT"
- "SMAX_COEFFICIENT"

### CODE REVIEW

File: applications/GeoMechanicsApplication/geo_mechanics_application.cpp [hunk 1]
```diff
@@ -341,6 +341,19 @@ void KratosGeoMechanicsApplication::Register() {
...
+    KRATOS_REGISTER_VARIABLE( AIR_HUMIDITY )
+    KRATOS_REGISTER_VARIABLE( PRECIPITATION )
+    KRATOS_REGISTER_VARIABLE( WIND_SPEED )
+    KRATOS_REGISTER_VARIABLE( A1_COEFFICIENT )
+    KRATOS_REGISTER_VARIABLE( A2_COEFFICIENT )
+    KRATOS_REGISTER_VARIABLE( A3_COEFFICIENT )
+    KRATOS_REGISTER_VARIABLE( ALPHA_COEFFICIENT )
+    KRATOS_REGISTER_VARIABLE( QF_COEFFICIENT )
+    KRATOS_REGISTER_VARIABLE( SMIN_COEFFICIENT )
+    KRATOS_REGISTER_VARIABLE( SMAX_COEFFICIENT )
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/geo_mechanics_application.cpp:356]: Just a general question: do we need to mind these names as well (to not run into the same issue as with the duplicate element names?)

Comment (rfaasse) [inline applications/GeoMechanicsApplication/geo_mechanics_application.cpp:356]: To me it's not directly clear what these coefficient mean and that they are related to micro-climate. Do we need any more clarification in the name, or is this enough?

**Review by avdg81 — DISMISSED** [review]
Very clear and clean PR. This is a good starter for the remainder of the micro-climate feature. I have no comments or suggestions.

**Review by rfaasse — COMMENTED** [review]
Thanks for adding these! I have some general questions/discussion points, but nothing blocking, so feel free to merge (e.g. tomorrow) when these points have been discussed and agreed if it is okay as is!

**Comment by mnabi** [comment]
@indigocoral 
Thank you for adding the variables. Very nice work.
Just a small remark, I see these variables are also added to "geo_mechanics_python_application.cp" in the original branch, but they are missing here. I am not sure about the effect, but I think it is for communication with python?

File: applications/GeoMechanicsApplication/geo_mechanics_application.cpp (hunk 1 above)
Comment (indigocoral) [reply applications/GeoMechanicsApplication/geo_mechanics_application.cpp:356]: As discussed with the team, a comment is added above the variables to indicate they are for Micro-climate, and for now they will stay as they are now until further discussions with CIMNE.

**Review by mnabi — APPROVED** [review]

**Review by mnabideltares — APPROVED** [review]

**Pull request merged**